packages/typespec-ts/test/integration/encode.spec.ts

```diff
@@ -15,56 +14,230 @@ describe("EncodeDurationClient Rest Client", () => {
+            input: "P40D"
+          }
+        });
+        assert.strictEqual(result.status, "204");
+      } catch (err) {
+        assert.fail(err as string);
+      }
+    });
+
+    it(`should get iso8601 query`, async () => {
+      try {
+        const result = await client.path(`/encode/duration/query/iso8601`).get({
+          queryParameters: {
+            input: "P40D"
+          }
+        });
+        assert.strictEqual(result.status, "204");
+      } catch (err) {
+        assert.fail(err as string);
+      }
+    });
+
+    it(`should get int32-seconds query`, async () => {
+      try {
+        const result = await client
+          .path(`/encode/duration/query/int32-seconds`)
+          .get({
+            queryParameters: {
+              input: 36
+            }
+          });
+        assert.strictEqual(result.status, "204");
+      } catch (err) {
+        assert.fail(err as string);
+      }
+    });
+
+    it(`should get int32-seconds-array query`, async () => {
+      try {
+        const result = await client
+          .path(`/encode/duration/query/int32-seconds-array`)
+          .get({
+            queryParameters: {
+              input: [36, 47]
+            },
+            skipUrlEncoding: true
+          });
+        assert.strictEqual(result.status, "204");
+      } catch (err) {
+        assert.fail(err as string);
+      }
+    });
+
+    it(`should get float-seconds query`, async () => {
       try {
         const result = await client
-          .path(`/encode/duration/query/${params.type}` as any)
+          .path(`/encode/duration/query/float-seconds`)
           .get({
             queryParameters: {
-              input: params.defaultValue
+              input: 35.621
+            }
+          });
+        assert.strictEqual(result.status, "204");
+      } catch (err) {
+        assert.fail(err as string);
+      }
+    });
+  });
+
+  describe("header", () => {
+    it(`should get default header`, async () => {
+      try {
+        const result = await client
+          .path(`/encode/duration/header/default`)
+          .get({
+            headers: {
+              duration: "P40D"
+            }
+          });
+        assert.strictEqual(result.status, "204");
+      } catch (err) {
+        assert.fail(err as string);
+      }
+    });
+
+    it(`should get iso8601 header`, async () => {
+      try {
+        const result = await client
+          .path(`/encode/duration/header/iso8601`)
+          .get({
+            headers: {
+              duration: "P40D"
+            }
+          });
+        assert.strictEqual(result.status, "204");
+      } catch (err) {
+        assert.fail(err as string);
+      }
+    });
+
+    it(`should get iso8601-array header`, async () => {
+      try {
+        const result = await client
+          .path(`/encode/duration/header/iso8601-array`)
+          .get({
+            headers: {
+              duration: ["P40D", "P50D"] as any
```

Comment:
@qiaozha I notice that `RawHttpHeadersInput` only allows for `string|number|boolean` which means there would exist type inference conflict for this property.

https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/core/core-rest-pipeline/src/interfaces.ts#L12-L15

I feel like `as any` is just a walkaround and would suggest to update the `RawHttpHeadersInput` header to allow some values like `string[]` or also build a helper function for the default csv format.

How do you think?

Reply:
Don't think we should change the RawHttpHeadersInput to add string[] type. let's add a helper for it.

Reply:
Sure I'll apply that change.